bot/cogs/giveaway.py

```diff
@@ -0,0 +1,171 @@
+import logging
+from datetime import datetime, timedelta, timezone
+from typing import Optional
+
+import discord
+from discord import app_commands, Interaction
+from discord.ext import commands, tasks
+
+from bot.bot import WarnetBot
+from bot.config import GiveawayConfig, GUILD_ID
+
+logger = logging.getLogger(__name__)
+
+
+@commands.guild_only()
+class Giveaway(commands.GroupCog, group_name='warnet-ga'):
+    def __init__(self, bot: WarnetBot) -> None:
+        self.bot = bot
+        self.db_pool = bot.get_db_pool()
+
+    @commands.Cog.listener()
+    async def on_connect(self) -> None:
+        self._check_blacklist_ga.start()
+
+    @app_commands.command(name='blacklist', description='Blacklist a user from giveaway')
+    @app_commands.describe(
+        amount='Giveaway prizes. Example: 50000',
+        winner='Winner of the giveaway. Example: 1234567890,0987654321',
+        ghosting='Member who do not claim the giveaway. Example: 1234567890,0987654321',
+    )
+    async def add_giveaway_blacklist(
+        self,
+        interaction: Interaction,
+        amount: int,
+        winner: str,
+        ghosting: Optional[str] = None,
+    ) -> None:
+        await interaction.response.defer()
+        if not interaction.user.guild_permissions.administrator:
+            return await interaction.followup.send('You are not an admin', ephemeral=True)
+
+        winnerss: list[discord.Member] = []
+        winners = winner.split(',')
+        for win in winners:
+            user = interaction.guild.get_member(int(win))
+            if user is None:
+                return await interaction.followup.send(f'User {win} not found', ephemeral=True)
+            winnerss.append(user)
+
+        ghostss: list[discord.Member] = []
+        if ghosting:
+            ghosts = ghosting.split(',')
+            for ghost in ghosts:
+                user = interaction.guild.get_member(ghost)
+                if user is None:
+                    return await interaction.followup.send(
+                        f'User {ghost} not found', ephemeral=True
+                    )
```

Comment:
Can be refactored like this
```suggestion
            for (ghosts := ghosting.split(',')) in ghosts:
                if (user := interaction.guild.get_member(ghost)):
                    return await interaction.followup.send(
                        f'User {ghost} not found', ephemeral=True
                    )
```